packages/block-editor/src/hooks/font-size.js

```diff
@@ -129,9 +120,20 @@ export function FontSizeEdit( props ) {
 		} );
 	};
 
-	return (
-		<FontSizePicker value={ fontSizeObject.size } onChange={ onChange } />
+	if ( isDisabled ) {
+		return null;
+	}
+
+	const fontSizeObject = getFontSize(
+		fontSizes,
+		fontSize,
+		style?.typography?.fontSize
 	);
+
+	const fontSizeValue =
+		fontSizeObject?.size || style?.typography?.fontSize || fontSize;
+
+	return <FontSizePicker onChange={ onChange } value={ fontSizeValue } />;
```

Comment:
Is this mapping necessary here? Wouldn't it be something to put inside the adapter function?

Reply:
> Is this mapping necessary here?

@gziolo I believe so. `style.typography.fontSize` is very specific to the `hooks/` part of the editor. After testing, it appears that `fontSizeValue` my change depending on values defined in the `theme.json` setup. If that's correct,  the prop "remap" feels correct in this `hooks/` area (vs. in the components package).